workflow/scripts/plot-volcano.R

```diff
@@ -0,0 +1,233 @@
+ 
+pval_transcripts <- sleuth_load(snakemake@input[["pvals"]])
+conditions <- read.table(snakemake@input[["samples"]])
+matrix <- read.table(snakemake@input[["matrix"]], quote="\"", check.names = F, stringsAsFactors=F, header = TRUE) 
+
+# #pval_transcripts <- sleuth_load("results/sleuth/diffexp/model_X.transcripts.diffexp.rds")
+# #conditions <- read.table("results/sleuth/samples.tsv")
+# #matrix <- read.table("results/tables/tpm-matrix/model_X.tpm-matrix.tsv", quote="\"", check.names = F, stringsAsFactors=F, header = TRUE)
+
+colnames(matrix)[1] = "target_id"
+matrix <- as_tibble(matrix)
+
+levels_condition <- levels(conditions$condition)
+
+if(length(levels_condition)!=2) {
+  print("Building volcano plot failed! There are too much different conditions!")
+} else {
+  condition_a_name <- levels_condition[1]
+  condition_b_name <- levels_condition[2]
+
+  condition_a <- as.character(factor(conditions$sample[conditions$condition == condition_a_name]))
+  condition_b <- as.character(factor(conditions$sample[conditions$condition == condition_b_name]))
+
+  samples <- as_tibble(matrix, header = TRUE)
+
+  samples_condition_a <- samples %>%
+    select(condition_a) 
+
+  samples_condition_b <- samples %>%
+    select(condition_b)
+
+#}
+ 
+
+ 
+  fold_change <- tibble(target_id=samples$target_id, ext_gene=samples$ext_gene, 
+                        FC=rowSums(samples_condition_a)/rowSums(samples_condition_b),
+                        FC_reverse=rowSums(samples_condition_b)/rowSums(samples_condition_a))
+  p_vals <- pval_transcripts %>%
+    select(target_id, ext_gene, pval, qval)
+    
+  log2FC <- fold_change %>%
+    select(target_id, ext_gene, log2FC=FC) %>%
+    mutate(log2FC=log2(log2FC)) 
+  
+  plot_data <- full_join(log2FC, p_vals, by=c("target_id", "ext_gene")) %>%
+    mutate(pval=-log10(pval))
+  
+  # plot_data <- plot_data[is.finite(plot_data$log2FC) && is.finite(plot_data$pval) && is.finite(plot_data$qval)]
+
+  # print(plot_data$log2FC)
+  # print(plot_data$pval)
+  #plot_data <- na.omit(plot_data)
+  # plot_data <- plot_data%>%mutate(sapply(c(log2FC,pval,qval),rm_nan_inf(x)))
+  # rm_nan_inf <- function(x){
+  #   plot_data$x[!(is.finite(plot_data$x))] <- 0
+  #   plot_data$x[is.na(plot_data$x)] <- 0
+  #   plot_data$x[is.nan(plot_data$x)] <- 0
+  # }
+  
+  # plot_data <- plot_data[is.finite(plot_data$log2FC)]
+ plot_data$log2FC[!(is.finite(plot_data$log2FC))] <- 0
+ plot_data$log2FC[is.na(plot_data$log2FC)] <- 0
+ plot_data$log2FC[is.nan(plot_data$log2FC)] <- 0
+
+  # plot_data <- plot_data[is.finite(plot_data$pval)]
+ plot_data$pval[!(is.finite(plot_data$pval))] <- 0
+ plot_data$pval[is.na(plot_data$pval)] <- 0
+ plot_data$pval[is.nan(plot_data$pval)] <- 0
+
+  # plot_data <- plot_data[is.finite(plot_data$qval)]
+ plot_data$qval[!(is.finite(plot_data$qval))] <- 0
+ plot_data$qval[is.na(plot_data$qval)] <- 0
+ plot_data$qval[is.nan(plot_data$qval)] <- 0
```

Comment:
I think we might also achieve this more elegantly with piping `plot_data` through some [`drop_na()`](https://tidyr.tidyverse.org/reference/drop_na.html) and some [`filter()`](https://dplyr.tidyverse.org/reference/filter.html) instances for the `Inf` values. Although we should also double-check, where these values come from in the first place, to make sure it's not a bug elsewhere. This is probably easiest done looking at actual data from an example run together.